src/components/code-block/index.js

```diff
@@ -0,0 +1,102 @@
+import { useState, useMemo, useRef, useEffect } from 'preact/hooks';
+import { Link } from 'preact-router';
+import cx from '../../lib/cx';
+import PrismWorker from 'workerize-loader?name=prism.[hash:5]!./prism.worker';
+
+const { highlight } = PRERENDER ? require('./prism.worker') : new PrismWorker();
+
+function useFuture(initializer, params) {
```

Comment:
This hook abstracts away the complicated parts of using a maybe-promise API like `highlight()`. This means during SSR, hydration and cached navigation, the first render can be fully syntax-highted synchronously without having to roundtrip through the Worker and cause a second render.